crates/types/src/invocation.rs

```diff
@@ -732,6 +752,35 @@ impl From<TraceId> for TraceIdDef {
     }
 }
 
+/// Defines how to query the invocation
+#[derive(Debug, Clone, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+pub enum InvocationQuery {
+    Invocation(InvocationId),
+    Workflow(ServiceId),
+}
```

Comment:
The docs don't really tell me what this enum is for and how the variants differ. Maybe you can add a bit more details to why this enum exists.